ivy/functional/frontends/tensorflow/tensor.py

```diff
@@ -228,6 +228,115 @@ def __iter__(self):
             yield self[i]
 
 
+class TensorShape:
+    # TODO: there are still some methods that may need implementing
+
+    def __init__(self, dims):
+        self._ivy_shape = dims if isinstance(dims, ivy.Shape) else ivy.shape(dims)
+        self._dims = self._ivy_shape._shape
```

Comment:
I would suggest keeping only `_dims` here and inferring the ivy shape in the corresponding property.
Other than that, looks good!

Reply:
Sure, thanks Anna!